nixos/doc/manual/development/option-types.section.md

```diff
@@ -326,7 +326,14 @@ Composed types are types that take a type as parameter. `listOf
 `types.uniq` *`t`*
 
 :   Ensures that type *`t`* cannot be merged. It is used to ensure option
-    definitions are declared only once.
+    definitions are provided only once.
+
+    ::: {.warning}
+    Deep type checks will not be performed. If you need `t` to check attribute
+    values or list items, use `types.unique` instead.
+    A future version of the module system may change `types.uniq` to match
+    `types.unique`'s behavior; to also check those lazy substructures.
```

Comment:
Not a fan of intentionally different behavior for two functions that originally had the same behavior. I don't think there's a good reason these should be distinguished like that.

I'd be in favor of making both `types.uniq` and `types.unique` behave in the stricter way. Really it not being strict before should be considered a bug.

Reply:
> making both `types.uniq` and `types.unique` behave in the stricter way. Really it not being strict before should be considered a bug.

Done and agreed.